src/Data/YAML/Dumper.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE Safe              #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+
+module Data.YAML.Dumper
+    ( dumpYAML
+    , dumpYAML'
+    , dumpEvents
+    ) where
+
+import           Data.YAML.Event.Internal as YE
+import           Data.YAML.Internal       as YI
+import           Data.YAML.Schema         as YS
+import           Data.YAML.Event.Writer   (writeEvents)
+
+import qualified Data.ByteString.Lazy     as BS.L
+import           Data.Map                 as Map
+import qualified Data.Text                as T
+
+
+type EvList = [Either String Event]
+type Node2EvList = [Node ()] -> EvList
+
+-- | Dump YAML Nodes using specified UTF encoding to a lazy 'BS.L.ByteString'
```

Comment:
As I don't think this is obvious it should be explained in the docstring that each Node is emitted as individual "YAML document"